server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -754,6 +758,8 @@ public void onResponse(IngestPipelinesExecutionResult result) {
                                         if (result.shouldKeep == false) {
                                             onDropped.accept(slot);
                                         }
+                                        assert firstPipeline != null;
+                                        firstPipeline.getPipelineMetrics().postIngest(indexRequest.ramBytesUsed());
```

Comment:
To avoid setting this value if the document is dropped, we can just add an `else` clause to the preceding `if`  clause and put the postIngest call in it.